### PR TITLE
TBB: enable RML, do not use text based .so files

### PR DESCRIPTION
**Pull request by extrowerk**
The LIBRARY_PATH needs some love in TEST().
Can somebody help here?

### CODE REVIEW

File: dev-cpp/tbb/patches/tbb-2018.5.patchset [hunk 1]
```diff
@@ -520,5 +500,97 @@ index dcf856a..baa9abd 100644
...
+--- a/build/Makefile.rml
++++ b/build/Makefile.rml
+@@ -112,8 +112,12 @@ $(RML.DLL): $(RML_TBB_DEP) $(RML_SERVER.OBJ) $(RML.RES) $(RML_NO_VERSION.DLL) $(
+ 
+ ifneq (,$(RML_NO_VERSION.DLL))
+ $(RML_NO_VERSION.DLL):
++ifeq (haiku,$(tbb_os))
++	ln -s $(RML.DLL) $(RML_NO_VERSION.DLL)
++else
+ 	echo "INPUT ($(RML.DLL))" > $(RML_NO_VERSION.DLL)
```
Comment (alaviss) [inline dev-cpp/tbb/patches/tbb-2018.5.patchset]: This shouldn't be a concern to us. The text files are `ld scripts`, and only used during build. Our dynamic loader should have used the `SONAME` instead, if not, chances are `SONAME` weren't set for the library when building on Haiku.

Comment (extrowerk) [reply dev-cpp/tbb/patches/tbb-2018.5.patchset]: The tests fails with runtime_loader complaining about the contents, thats why i did this.
Do you mean a non-set SONAME could be the culprit?

**Comment by alaviss** [comment]
Yes. Our linker should have loaded `libtbb*.so.2` directly since it's the `SONAME` (checked on Linux).

Maybe you were missing the `-Wl,--soname=` link flag. Look at `build/linux.gcc.inc` for an example.


-------- Original Message --------

**Comment by extrowerk** [comment]
We have literally the same part in haiku_gcc.inc (it is based on linux_gcc.inc, but trimmed as the TBB guys wanted).
So i still doesn't really understand, whats going on.

**Comment by alaviss** [comment]
Got this lurking in the logs while building:

```
hostname: invalid option -- 's'
Try 'hostname --help' for more information.
../../build/version_info_linux.sh: line 29: getconf: command not found
```

This might be the culprit.

File: dev-cpp/tbb/patches/tbb-2018.5.patchset (hunk 1 above)
Comment (alaviss) [reply dev-cpp/tbb/patches/tbb-2018.5.patchset]: The tests run fine without that patch. Got a compiler segmentation fault however:

```
g++ -o test_eh_algorithms.o -c -MMD -DDO_ITT_NOTIFY -g -O2 -DUSE_PTHREAD -m64 -mrtm   -DTEST_USES_TBB=1 -Wall -Wshadow -Wcast-qual -Woverloaded-virtual -Wnon-virtual-dtor -Wextra    -I../../src -I../../src/rml/include -I../../include -I. -include ../../src/test/harness_preload.h ../../src/test/test_eh_algorithms.cpp
../../src/test/test_eh_algorithms.cpp: In function 'void TestParallelLoopAux() [with ParForBody = SimpleParForBody; Partitioner = const tbb::simple_partitioner]':
../../src/test/test_eh_algorithms.cpp:162:6: internal compiler error: Segmentation violation
 void TestParallelLoopAux() {
      ^~~~~~~~~~~~~~~~~~~
libbacktrace could not find executable to open
Please submit a full bug report,
with preprocessed source if appropriate.
See <https://gcc.gnu.org/bugs/> for instructions.
../../build/common_rules.inc:86: recipe for target 'test_eh_algorithms.o' failed
make[1]: *** [test_eh_algorithms.o] Error 1
make[1]: Leaving directory '/sources/tbb-2018_U5/build/haiku_intel64_gcc_cc7.3.0_libc_kernel1_release'
Makefile:44: recipe for target 'test' failed
```

File: dev-cpp/tbb/patches/tbb-2019.patchset [hunk 2]
```diff
@@ -520,5 +520,99 @@ index dcf856a..baa9abd 100644
...
-2.15.0
+2.19.0
+
+
+
+
+From 8f22979d461865316e05515fcbb56b54e8a74271 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Zolt=C3=A1n=20Mizsei?= <zmizsei@extrowerk.com>
+Date: Sat, 1 Sep 2018 22:49:59 +0200
+Subject: Haiku runtime_loader cant use textfiles, use symlinks instead
```
Comment (alaviss) [inline dev-cpp/tbb/patches/tbb-2019.patchset]: The tests run fine without that patch. Got a compiler segmentation fault however:

```
g++ -o test_eh_algorithms.o -c -MMD -DDO_ITT_NOTIFY -g -O2 -DUSE_PTHREAD -m64 -mrtm   -DTEST_USES_TBB=1 -Wall -Wshadow -Wcast-qual -Woverloaded-virtual -Wnon-virtual-dtor -Wextra    -I../../src -I../../src/rml/include -I../../include -I. -include ../../src/test/harness_preload.h ../../src/test/test_eh_algorithms.cpp
../../src/test/test_eh_algorithms.cpp: In function 'void TestParallelLoopAux() [with ParForBody = SimpleParForBody; Partitioner = const tbb::simple_partitioner]':
../../src/test/test_eh_algorithms.cpp:162:6: internal compiler error: Segmentation violation
 void TestParallelLoopAux() {
      ^~~~~~~~~~~~~~~~~~~
libbacktrace could not find executable to open
Please submit a full bug report,
with preprocessed source if appropriate.
See <https://gcc.gnu.org/bugs/> for instructions.
../../build/common_rules.inc:86: recipe for target 'test_eh_algorithms.o' failed
make[1]: *** [test_eh_algorithms.o] Error 1
make[1]: Leaving directory '/sources/tbb-2018_U5/build/haiku_intel64_gcc_cc7.3.0_libc_kernel1_release'
Makefile:44: recipe for target 'test' failed
```

**Comment by extrowerk** [comment]
tls fixed, let's continue this...

**Comment by extrowerk** [comment]
OMG

**Comment by extrowerk** [comment]
Now thats never happened before, what should i do now?

**Comment by korli** [comment]
Why enabling RML?

**Comment by extrowerk** [comment]
@korli I can't answer it anymore, it was long time ago, but it is easy to disable it.

**Pull request closed without merging**